system_tests/contract_tx_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package arbtest
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/offchainlabs/nitro/arbnode"
+	"github.com/offchainlabs/nitro/arbos"
+	"github.com/offchainlabs/nitro/arbstate"
+	"github.com/offchainlabs/nitro/util/arbmath"
+)
+
+func TestContractTxDeploy(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	nodeconfig := arbnode.ConfigDefaultL2Test()
+	l2info, node, client := CreateTestL2WithConfig(t, ctx, nil, nodeconfig, false)
+	defer node.StopAndWait()
+
+	from := common.HexToAddress("0x123412341234")
+	TransferBalanceTo(t, "Faucet", from, big.NewInt(1e18), l2info, client, ctx)
+
+	var requestId common.Hash
+	var stateNonce uint64
+	runDeploy := func() *types.Receipt {
```

Comment:
erm.. this could just be a simple loop running a few times?

Reply:
Yep, the invocation of this was initially more complicated, but now it's just doing the same thing twice :sweat_smile: